Include/pycapi/pycapi_objimpl.h

```diff
@@ -0,0 +1,163 @@
+#ifndef Py_PYCAPI_MEM_H
+#define Py_PYCAPI_MEM_H
+#ifdef __cplusplus
+extern "C" {
+#endif
+
```

Comment:
I wanted to add:

```
#ifdef Py_LIMITED_API
#  error "Py_LIMITED_API must not be defined to include this header"
#endif
```

But this header should be include if Py_LIMITED_API is defined with a value.

Reply:
If you don't move PyObject_Calloc (because it's the stable ABI in 3.5+), then you can add that guard to the unstable ABI header.